wdl2cwl/main.py

```diff
@@ -433,28 +433,37 @@ def get_outdir_requirement(
                 isinstance(outdir, (WDL.Expr.Apply, WDL.Expr.String))
                 and outdir.literal is None
             ):
+                # runtime["disks"] usually have strings chars like local, HDD, SDD that are not needed by CWL
                 if (
                     isinstance(outdir, WDL.Expr.Apply)
                     and not outdir.function_name == "_add"
                 ):
+                    # If it contains an apply expr we don't want to process the _add function
+                    # that concatenates it to the chars in the string
                     expr_str = self.get_expr(outdir)
+                    # if the expr_str returns an integer that is declared as a static variable
+                    # return the result from the multiplication with 1024
                     return (
                         f"$(({expr_str}) * 1024)"
                         if not expr_str.isdigit()
                         else int(expr_str) * 1024
                     )
+                # apply exprs contain arguments and strings contain parts both are lists
+                # they could contain expressions that represent the runtime disk
                 list_object = (
                     outdir.arguments if hasattr(outdir, "arguments") else outdir.parts  # type: ignore
                 )
                 for obj in list_object:
                     if isinstance(
                         obj, (WDL.Expr.Get, WDL.Expr.Apply, WDL.Expr.Placeholder)
                     ):
+                        # avoid python strings only WDL expressions are handled.
                         return self.get_outdir_requirement(obj)  # type: ignore
             elif isinstance(outdir, (WDL.Expr.Get, WDL.Expr.Placeholder)):
                 expr = self.get_expr(outdir)
                 expr_str = expr
                 if isinstance(outdir, (WDL.Expr.Placeholder)):
+                    # expr returns a string that contains $(). This has to be removed.
```

Comment:
`get_expr()` that returns a string wrapped in `$(` and `)`, yes?

Maybe `get_expr()` shouldn't and we just add the `$(` and `)` where it makes sense? Would be less mysterious :-)

Reply:
Awesome Idea